atomicfu/src/concurrentTest/kotlin/kotlinx/atomicfu/locks/TimedParkingTest.kt

```diff
@@ -0,0 +1,231 @@
+package kotlinx.atomicfu.locks
+
+import kotlinx.atomicfu.atomic
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlin.time.measureTime
+import kotlin.IllegalStateException
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.nanoseconds
+
+class TimedParkingTest {
+
+    @Test
+    fun testNanosFirstUnpark400() = retry(3) {
```

Comment:
Reliability of these tests worries me. 100ms slack feels huge, but in practice we can accidentally exceed it and got a test failure (and I guess we're already doing that given the test failures on Windows).

For deadline-specific scenarios, I don't think we need to unpark a future from the main thread at all, and we should only verify that it was parked for at least `timeout`-time-units.

For first-unpark-specific scenarios, we can't get away from verifying that the `park` was actually "interrupted" by the `unpark` call, but maybe we can increase the slack to a few minutes, instead of 100ms?

Reply:
Agreed, I tried to fix it with a few retries. But the tests on windows stay unreliable. Thanks for the suggestions!